lib/Sema/CSFix.cpp

```diff
@@ -1168,3 +1168,29 @@ SpecifyClosureReturnType::create(ConstraintSystem &cs,
                                  ConstraintLocator *locator) {
   return new (cs.getAllocator()) SpecifyClosureReturnType(cs, locator);
 }
+
+bool SpecifyObjectLiteralTypeImport::diagnose(bool asNote) const {
+  auto &cs = getConstraintSystem();
+  UnableToInferProtocolLiteralType failure(cs, getLocator());
+  return failure.diagnose(asNote);
+}
+
+SpecifyObjectLiteralTypeImport *
+SpecifyObjectLiteralTypeImport::attempt(ConstraintSystem &cs,
+                                        ConstraintLocator *locator) {
+  auto *anchor = locator->getAnchor();
+  if (!isa<ObjectLiteralExpr>(anchor))
+    return nullptr;
+  
+  auto protocol = TypeChecker::getLiteralProtocol(cs.getASTContext(), anchor);
+  if (!protocol)
+    return nullptr;
+  
+  if (TypeChecker::getDefaultType(protocol, cs.DC))
+    return nullptr;
+
+  if (cs.getContextualType())
```

Comment:
I think you should do this before the calls to `TypeChecker::` since they are more expensive.

Reply:
That makes sense, Thanks :)